packages/plugins/accounts-manager/src/index.ts

```diff
@@ -82,4 +82,54 @@ export default class AccountsManager {
       this.logger.error(__("Error funding accounts"), err.message || err);
     }
   }
+
+  async getCoinbaseAddress(web3) {
+    const findAccountWithMostFunds = async () => {
+      const accounts = await web3.eth.getAccounts();
+      let highestBalance = {
+        balance: web3.utils.toBN(0),
+        account: ""
+      };
+      for (const account of accounts) {
+        // eslint-disable-next-line no-await-in-loop
+        let balance = await web3.eth.getBalance(account);
+        balance = web3.utils.toBN(balance);
+        if (balance.gt(highestBalance.balance)) {
+          highestBalance = { balance, account };
+        }
+      }
+      return highestBalance.account;
+    };
+
+    const findAlternativeCoinbase = async () => {
+      try {
+        return findAccountWithMostFunds();
+      } catch (err) {
+        throw new Error(`Error getting coinbase address: ${err.message || err}`);
+      }
+    };
```

Comment:
I think `findAccountWithMostFunds` and `findAlternativeCoinBase` can be wonderfully defined **outside** of `getCoinbaseAddress`.

IOW: we probably don't want to redefine these functions, every time `getCoinbaseAddress` is called..

Reply:
Good point, moved!